src/DeviceInput/InputDevices/webDeviceInputSystem.ts

```diff
@@ -482,6 +482,11 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
                 deviceEvent.deviceType = deviceType;
                 deviceEvent.deviceSlot = deviceSlot;
 
+                deviceEvent.inputIndex = evt.button + 2;
+                deviceEvent.previousState = previousButton;
+                deviceEvent.currentState = pointer[evt.button + 2];

```

Comment:
you could use `deviceEvent.inputIndex` instead of adding 2 again

Reply:
Added for Up and Down